master/pkg/model/logging_config.go

```diff
@@ -0,0 +1,100 @@
+package model
+
+import (
+	"encoding/json"
+	"io/ioutil"
+
+	"github.com/pkg/errors"
+
+	"github.com/determined-ai/determined/master/pkg/union"
+)
+
+// LoggingConfig configure logging for tasks (currently only trials) in Determined.
+type LoggingConfig struct {
+	DefaultLoggingConfig *DefaultLoggingConfig `union:"backend,default" json:"-"`
+	ElasticLoggingConfig *ElasticLoggingConfig `union:"backend,elastic" json:"-"`
+}
+
+// Resolve resolves the parts of the TaskContainerDefaultsConfig that must be evaluated on
+// the master machine.
+func (c LoggingConfig) Resolve() error {
+	if c.ElasticLoggingConfig != nil {
+		err := c.ElasticLoggingConfig.Resolve()
+		if err != nil {
+			return err
+		}
+	}
+	return nil
+}
+
+// MarshalJSON serializes LoggingConfig.
+func (c LoggingConfig) MarshalJSON() ([]byte, error) {
+	return union.Marshal(c)
+}
+
+// UnmarshalJSON deserializes LoggingConfig.
+func (c *LoggingConfig) UnmarshalJSON(data []byte) error {
+	if err := union.Unmarshal(data, c); err != nil {
+		return err
+	}
+
+	type DefaultParser *LoggingConfig
+	return errors.Wrap(json.Unmarshal(data, DefaultParser(c)), "failed to parse logging options")
+}
+
+// DefaultLoggingConfig configure logging for tasks using Fluent+HTTP to the master.
+type DefaultLoggingConfig struct{}
+
+// ElasticLoggingConfig configure logging for tasks using Fluent+Elastic.
+type ElasticLoggingConfig struct {
+	Host     string                `json:"host"`
+	Port     int                   `json:"port"`
+	Security ElasitcSecurityConfig `json:"security"`
+}
+
+// Resolve resolves the parts of the ElasticLoggingConfig that must be evaluated on the
+// master machine.
+func (o *ElasticLoggingConfig) Resolve() error {
```

Comment:
non-blocking: This seems better placed as a method on `ElasticTLSConfig`. I presume you did it this way to avoid having an extra tiny method or two around just to get all the way down there? Not unreasonable, but I think I'd still prefer to keep it separated the other way.

Reply:
yeah that's exactly why I did it that way; sure I can change it.